torch/_inductor/codecache.py

```diff
@@ -56,40 +66,99 @@ def _compile_end():
 
 @functools.lru_cache(None)
 def cache_dir():
-    return os.environ.get(
+    cache_dir = os.environ.get(
         "TORCHINDUCTOR_CACHE_DIR",
         f"{tempfile.gettempdir()}/torchinductor_{getpass.getuser()}",
     )
+    os.makedirs(cache_dir, exist_ok=True)
+    return cache_dir
 
 
-class DiskCache:
-    @staticmethod
-    @functools.lru_cache(None)
-    def _subdir():
-        subdir = os.path.join(cache_dir(), "cached_tunings")
-        os.makedirs(subdir, exist_ok=True)
-        return subdir
+class PersistentCache:
+    def __init__(self):
+        self.local_cache_path = os.path.join(cache_dir(), "local_cache")
+        self.global_cache_path = config.global_cache_path
 
-    @staticmethod
-    @functools.lru_cache(4096)
-    def _read_file(path):
-        with open(path, "r") as fd:
-            return json.loads(fd.read())
+        if torch.cuda.is_available():
+            self.dinfo = repr(
+                torch.cuda.get_device_properties(torch.cuda.current_device())
+            )
 
-    def __init__(self, unique_name):
-        super().__init__()
-        self.unique_name = unique_name
+    def get_local_cache(self):
+        if not os.path.isfile(self.local_cache_path):
+            return {}
+        with open(self.local_cache_path, "r") as local_cache_file:
+            local_cache = json.load(local_cache_file)
+        return local_cache
 
-    def lookup(self, key: Any, generate: Callable[[], Any]):
+    def update_local_cache(self, local_cache):
+        counters["inductor"]["select_algorithm_autotune_local_cache_updated"] += 1
+        write_atomic(self.local_cache_path, json.dumps(local_cache, indent=4))
```

Comment:
I worry that this pattern of load/update/write will be `O(n^2)`, as the cache gets very big this will get more and more expensive.

Perhaps it is small enough it won't matter though.

Reply:
I'd expect the local cache to be almost always empty, or very close to empty, otherwise we aren't doing a very good job of updating the global cache.